bouncer/scripts/redirect.js

```diff
@@ -1,6 +1,27 @@
 'use strict';
 
-/** Return the settings object that the server injected into the page. */
+/**
+ * Configuration information for the client-side code that detects the best way
+ * to route the user to a URL with Hypothesis activated and specified
+ * annotations selected.
+ *
+ * This is rendered into Bouncer's interstitial page by the backend service.
+ *
+ * @typedef {Object} Settings
```

Comment:
:) - I was wondering if there is something similar for Python dict shapes, given the return type of `parse_document`, although perhaps that function should just return a class instead?

Reply:
Yeah, I’d usually opt for passing around some kind of class instance rather than a dictionary at that point. By the time you need to document what shape the thing is, I think it’s worth enforcing that shape in code. When it’s a flat structure without a lot of attached functionality, I’m a big fan of [`namedtuple`s](https://docs.python.org/2/library/collections.html#collections.namedtuple).